python/ray/data/_internal/dataset_logger.py

```diff
@@ -1,8 +1,107 @@
 import logging
 import os
+from typing import Tuple
 
 import ray
 from ray._private.ray_constants import LOGGER_FORMAT, LOGGER_LEVEL
+from ray.data.context import DataContext
+
+
+class UserCodeException(Exception):
```

Comment:
It would be great if we could make these top-level members of `ray.data`, so that users don't see the `ray.data._internal.dataset_logger` part.